File: Jellyfin.Plugin.HomeMessage/Web/js/@types/Message.ts
export interface Message {
  /**
   * The ID of the message.
   */
  Id: string;

  /**
   * The title of the message.
   */
  Title: string;

  /**
   * The message.
   */
  Text: string;

  /**
   * Whether or not the message is dismissible.
   */
  Dismissible: boolean;

  /**
   * The background color of the message.
   */
  BgColor: string;

  /**
   * The text color of the message.
   */
  TextColor: string;

  /**
   * The time when the message should be shown.
   */
  TimeStart?: number;

  /**
   * The time when the message should be hidden.
   */
  TimeEnd?: number;

  /**
   * Unix timestamp of when the message was created.
   */
  CreatedTime: number;
}

/**
 * Represents a message input.
 */
export interface MessageInput {
  Title: string;
  Text: string;
  Dismissible: boolean;
  BgColor: string;
  TextColor: string;
  TimeStart: number | null;
  TimeEnd: number | null;
}

File: Jellyfin.Plugin.HomeMessage/Models/Message.cs
using Jellyfin.Plugin.HomeMessage.Store;

namespace Jellyfin.Plugin.HomeMessage.Models;

/// <summary>
/// Represents a single message to display on the home page.
/// </summary>
public record Message(
    // The title of the message.
    string Title,
    // The message.
    string Text,
    // Whether or not the message is dismissible.
    bool Dismissible,
    // The background color of the message.
    string BgColor,
    // The text color of the message.
    string TextColor,
    // The time when the message should be shown.
    int? TimeStart,
    // The time when the message should be hidden.
    int? TimeEnd
) : IModel
{
    /// <inheritdoc />
    public string Id { get; set; } = string.Empty;

    /// <inheritdoc />
    public long CreatedTime { get; set; }
}
